### DISCUSSION TITLE
How to create lighthouse-agent image through source code？

### DISCUSSION THREAD

**Discussion by zcz-1020** [submariner-io/lighthouse#1028] [Q&A] [answered]
Recently, I wanted to create images of lighthouse-agent and lighthous-coredns from lighthouse source code. But it didn't work. How do you do this? I'm a novice at this.

Here's how I do it:
PC：win10 OS 
1. Download the source code of lighthouse.
2. Execute `docker buildx build --platform linux/amd64 -f package/Dockerfile.lighthouse-agent -t lighthouse:test.`

The following error is reported:
` 
 [builder 3/3] RUN make -C /go/src/github.com/submariner-io/lighthouse LOCAL_BUILD=1 bin/linux/amd64/lighthouse-agent:
#0 0.345 go mod download
#0 4.367 go: cloud.google.com/go/compute@v1.7.0: Get "https://proxy.golang.org/cloud.google.com/go/compute/@v/v1.7.0.mod": x509: certificate signed by unknown authority
#0 4.368 make: Leaving directory '/go/src/github.com/submariner-io/lighthouse'
#0 4.368 make: *** [/opt/shipyard/Makefile.inc:281: vendor/modules.txt] Error 1
`
I don't know if it's my wrong operation or the environment configuration on the computer is incorrect.

**Comment by skitt** [comment, marked as answer]
To build the Lighthouse images (or any image in any Submariner project), the supported command is

    make images

However we don’t test this on Windows; at minimum I suspect you’d have to run this from a WSL2 shell.
